apps/src/lib/kits/maker/boards/microBit/MicroBitBoard.js

```diff
@@ -156,6 +163,62 @@ export default class MicroBitBoard extends EventEmitter {
     return !!this.boardClient_.myPort;
   }
 
+  /**
+   * Displays  to demonstrate successful connection
+   * A square is drawn in a spiral and then a checkmark flashes 3 times
+   * on the board.
+   * @returns {Promise} resolved when the animation is done.
+   */
+  celebrateSuccessfulConnection() {
+    function makeSquare(ledScreen, delay, timeInterval) {
+      return new Promise(resolve => {
+        SQUARE_LEDS.forEach((ledPair, i) => {
+          setTimeout(
+            () => ledScreen.on(ledPair[0], ledPair[1]),
+            delay * (i + 1)
+          );
+        });
+        setTimeout(resolve, delay * SQUARE_LEDS.length + timeInterval);
+      });
+    }
+
+    function makeCheckMark(ledScreen, delay) {
+      return new Promise(resolve => {
+        CHECKMARK_LEDS.forEach(ledPair => {
+          setTimeout(() => ledScreen.on(ledPair[0], ledPair[1]));
+        });
+        setTimeout(resolve, delay);
+      });
+    }
+
+    function turnOffAllLeds(ledScreen, delay) {
+      return new Promise(resolve => {
+        ALL_LEDS.forEach(ledPair => {
+          setTimeout(() => ledScreen.off(ledPair[0], ledPair[1]));
+        });
+        setTimeout(resolve, delay);
+      });
+    }
+
+    this.initializeComponents().then(() => {
+      const ledScreen = new LedScreen({
+        mb: this.boardClient_
+      });
+
+      const timeInterval = 200;
+      const squareTimeInterval = 70;
+
+      return Promise.resolve()
+        .then(() => makeSquare(ledScreen, squareTimeInterval, timeInterval))
+        .then(() => turnOffAllLeds(ledScreen, timeInterval))
+        .then(() => makeCheckMark(ledScreen, timeInterval))
+        .then(() => turnOffAllLeds(ledScreen, timeInterval))
+        .then(() => makeCheckMark(ledScreen, timeInterval))
+        .then(() => turnOffAllLeds(ledScreen, timeInterval))
+        .then(() => makeCheckMark(ledScreen, timeInterval));
```

Comment:
should we call `turnOffAllLeds` again after we've drawn the last checkmark?

Reply:
We disconnect the M:B after 3 seconds so that the last checkmark persists a little longer which I thought was nice. But I can definitely call `turnOffAllLeds` so that the checkmarks flash for the same time interval. Lmk what you think. I'll ask Dan S as well. Thanks!

Reply:
ahhh, okay -- as long as disconnecting does the necessary cleanup so that the checkmark doesn't persist forever or leave the LEDs in a weird state, i think it's totally fine to leave as-is. whatever style you/dan prefer seems fine here!